src/containers/CodesplainAppBar.jsx

```diff
@@ -50,8 +59,33 @@ export class CodesplainAppBar extends Component {
   }
 
   componentDidMount() {
-    const { dispatch } = this.props;
-    dispatch(updateUserSnippets());
+    const { dispatch, token } = this.props;
+    const tokenCookie = cookie.load('token');
+
+    // Save token to state if it hasn't already been (by <Auth />)
+    if (!token && tokenCookie) {
+      dispatch(saveAccessToken(tokenCookie));
+    }
+
+    // If we have a token (thus are logged in), get user's info & save to state
+    if (tokenCookie) {
+      dispatch(fetchUserOrgs())
+        .then(({ data }) => {
+          // Dispatch orgs to state
+          data.forEach(org => dispatch(addOrg(org.login)));
+        })
+        .then(() => dispatch(fetchUserInfo()))
+        .then((res) => {
+          const { login: username, avatar_url: userAvatarURL } = res.data;
+          dispatch(saveUsername(username));
+          dispatch(setAvatarUrl(userAvatarURL));
+
+          // Add user's username to orgs list, and select it as default
+          dispatch(addOrg(username));
+          dispatch(switchOrg(username));
+        })
+        .then(() => dispatch(updateUserSnippets()));
+    }
```

Comment:
It would be beneficial to have a catch clause in case the token isn't valid. Specifically, it should remove the token from cookies and redirect to the home page